cse482spring2019/Crash
# GeoLocation

#### What I did
* Implemented a wrapper component `GeoLocator`
* Added `redux`, `immutable`, and a few related packages
* Implemented a dummy component `GeoLocatorView` to demonstrate how to use `GeoLocator`

#### How to verify it
* Run the app (`npm start` from `/app` and scan the QR code)
* The app should ask for permission to access location, and if you confirm it should display your location

#### How to use `GeoLocator`
`GeoLocator` is a function that, if called with a component as its argument, will return a component that has access to location data (and provides that location data to the argument component as the prop `location`).

For example, if I have a component called `LocationView` and I call define `const WrappedLocationView = GeoLocator(LocationView)`, I can use `<WrappedLocationView />` in a render function, and the `LocationView` component that was wrapped can use `this.props.location`.
Any props you provide to `<WrappedLocationView />` will be provided to `<LocationView />`. Additionally, if you call it with `<WrappedLocationView watchLocation />` (which passes a `watchLocation` prop with value `true`), the wrapped `LocationView` will be subscribed to regular location changes, as opposed to needing to request the location. If `watchLocation` is not provided or is set to false, `LocationView` will also be passed a prop `this.props.fetchLocation`, which is a function that returns nothing, but prompts the app to request the current location.

`this.props.location` is an object of the following structure:
```javascript
{
  timestamp: Date,
  error: null | any,
  coords: null | { longitude: number, latitude: number, accuracy: number }
}
```
Either `error` or `coords` will always be non-null (and non-undefined), but never both.
